### PR TITLE
feat: Add CakePHP 3+ Support

### PR DESCRIPTION
**Pull request by PROFeNoM**
### Description

Add support for CakePHP 3+ by replacing the hook arguments with the new CakePHP v3+ classes.

Documented in https://github.com/DataDog/documentation/pull/22569

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: src/Integrations/Integrations/CakePHP/CakePHPIntegration.php [hunk 1]
```diff
@@ -35,115 +33,16 @@ public function init()
         $integration = $this;
 
         // Since "Dispatcher" and "App" are common names, check for a CakePHP signature before loading
+        /*
         if (!defined('CAKE_CORE_INCLUDE_PATH')) {
             return self::NOT_AVAILABLE;
         }
+        */
```
Comment (bwoebi) [inline src/Integrations/Integrations/CakePHP/CakePHPIntegration.php:40]: I suppose you will want to continue checking for these for CakePHP V2 (i.e. if the class_exists check fails)?

File: src/Integrations/Integrations/CakePHP/V3/CakePHPIntegrationLoader.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        \DDTrace\trace_method('Cake\Error\Middleware\ErrorHandlerMiddleware', 'handleException', [
+            'instrument_when_limited' => 1,
+            'posthook' => function (SpanData $span, array $args) use ($integration) {
+                $integration->setError($integration->rootSpan, $args[0]);
+                $span->meta[Tag::COMPONENT] = CakePHPIntegration::NAME;
+                return false;
+            },
+        ]);
+
+        \DDTrace\trace_method('Cake\Http\Response', 'getStatusCode', [
```
Comment (bwoebi) [inline src/Integrations/Integrations/CakePHP/V3/CakePHPIntegrationLoader.php:87]: I suppose you copied from V2, but shouldn't we just use proper hook_method for these? I mean `return false` just unconditionally drops the span...

I think we can improve that one in all instances here.

Comment (PROFeNoM) [reply src/Integrations/Integrations/CakePHP/V3/CakePHPIntegrationLoader.php:87]: Absolutely :) Copy-pasted as you said and didn't pay attention to this; I'll change these

File: src/Integrations/Integrations/CakePHP/V3/CakePHPIntegrationLoader.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+use DDTrace\SpanData;
+use DDTrace\Tag;
+use DDTrace\Type;
+use DDTrace\Util\Normalizer;
+
+class CakePHPIntegrationLoader
+{
+    public function load($integration)
+    {
+        $integration->rootSpan = null;
```
Comment (bwoebi) [inline src/Integrations/Integrations/CakePHP/V3/CakePHPIntegrationLoader.php:19]: Can we please move away from caching rootSpan and rather refetch \DDTrace\root_span(), as more and more frameworks are also working under Swoole (where the root span isn't global, but may change with the executing fiber).

**Review by bwoebi — APPROVED** [review]
Looks good to me :-)

**Pull request merged**